tools/webpack/experimental.config.js

```diff
@@ -0,0 +1,423 @@
+
+const WORDPRESS_NAMESPACE = '@wordpress/';
+const BUNDLED_PACKAGES = [ '@wordpress/icons', '@wordpress/interface' ];
+
+const GUTENBERG_PACKAGES = Object.keys( dependencies )
+	.filter(
+		( packageName ) =>
+			! BUNDLED_PACKAGES.includes( packageName ) &&
+			packageName.startsWith( WORDPRESS_NAMESPACE ) &&
+			! packageName.startsWith( WORDPRESS_NAMESPACE + 'react-native' )
+	)
+	.map( ( packageName ) => packageName.replace( WORDPRESS_NAMESPACE, '' ) );
+
+const EXPORT_DEFAULT_PACKAGES = [
+	'api-fetch',
+	'deprecated',
+	'dom-ready',
+	'redux-routine',
+	'token-list',
+	'server-side-render',
+	'shortcode',
+	'warning',
+];
+
+const BLOCK_LIBRARY_SOURCE_PATH = join(
+	__dirname,
+	'..',
+	'..',
+	'packages',
+	'block-library',
+	'src'
+);
+
+function createStyleJSEntry( importPath ) {
+	return {
+		import: importPath,
+		filename: '[name].css-entry.js',
+	};
+}
+
+function getEntries() {
+	const entry = {
+		'block-editor/default-editor-styles': createStyleJSEntry(
+			'./packages/block-editor/src/default-editor-styles.scss'
+		),
+		'block-library/reset': createStyleJSEntry(
+			join( BLOCK_LIBRARY_SOURCE_PATH, 'reset.scss' )
+		),
+		'block-library/editor': createStyleJSEntry(
+			join( BLOCK_LIBRARY_SOURCE_PATH, 'editor.scss' )
+		),
+		'block-library/theme': createStyleJSEntry(
+			join( BLOCK_LIBRARY_SOURCE_PATH, 'theme.scss' )
+		),
+	};
+
+	for ( const packageName of GUTENBERG_PACKAGES ) {
+		entry[ `${ packageName }/index` ] = {
+			import: [ `./packages/${ packageName }` ],
+			library: {
+				name: [ 'wp', camelCaseDash( packageName ) ],
+				type: 'window',
+				export: EXPORT_DEFAULT_PACKAGES.includes( packageName )
+					? 'default'
+					: undefined,
+			},
+		};
+
+		const stylePath = `./packages/${ packageName }/src/style.scss`;
+		if ( existsSync( stylePath ) ) {
+			entry[ `${ packageName }/index` ].import.unshift( stylePath );
+		}
+	}
+
+	const blockNames = readdirSync( BLOCK_LIBRARY_SOURCE_PATH, {
+		withFileTypes: true,
+	} )
+		.filter( ( dirent ) => dirent.isDirectory() )
+		.map( ( dirent ) => dirent.name );
+
+	for ( const blockName of blockNames ) {
+		const scriptPath = join(
+			BLOCK_LIBRARY_SOURCE_PATH,
+			blockName,
+			'view.js'
+		);
+		if ( existsSync( scriptPath ) ) {
+			entry[ `block-library/blocks/${ blockName }/view` ] = scriptPath;
+		}
+
+		[ 'style', 'editor', 'theme' ].forEach( ( styleFile ) => {
+			const stylePath = join(
+				BLOCK_LIBRARY_SOURCE_PATH,
+				blockName,
+				`${ styleFile }.scss`
+			);
+			if ( existsSync( stylePath ) ) {
+				entry[
+					`block-library/blocks/${ blockName }/${ styleFile }`
+				] = createStyleJSEntry( stylePath );
+			}
+		} );
+	}
+
+	return entry;
+}
+
+threadLoader.warmup(
+	{
+		poolTimeout: Infinity,
+	},
+	[ require.resolve( 'babel-loader' ), '@wordpress/babel-preset-default' ]
+);
+
+module.exports = {
+	mode,
+	devtool:
+		devtool ||
+		( mode === 'production' ? false : 'eval-cheap-module-source-map' ),
```

Comment:
For some reasons, source maps for CSS only works if we use `source-map` here. I guess it could be an opt-in thing for now if folks really need source maps for them.